File: src/test/resources/integration_tests/javascript/test_verticle.js

var container = require("vertx/container")
var vertx = require("vertx");
var vertxTests = require("vertx_tests");
var vassert = require("vertx_assert");

var type = container.config['type']

switch (type) {
  case 'fail':
    vassert.fail("failed");
    break;
  case 'assert_fail':
    vassert.assertEquals('foo', 'bar');
    break;
  case 'assert_ok':
    vassert.assertEquals('foo', 'foo');
    vassert.testComplete();
}

File: src/test/resources/integration_tests/python/test_verticle.py
from org.vertx.testtools import VertxAssert
import vertx_tests
from core.event_bus import EventBus
import vertx

type = vertx.config()['type']

if type == 'fail':
    VertxAssert.fail('failed')
elif type == 'assert_fail':
    VertxAssert.assertEquals('foo', 'bar')
elif type == 'assert_ok':
    VertxAssert.assertEquals('foo', 'foo')
    VertxAssert.testComplete()


